### PR TITLE
SNOW-1032347: update comment of session.sql for lazy evaluation

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/session.py [hunk 1]
```diff
@@ -1913,8 +1913,10 @@ def generator(
     def sql(self, query: str, params: Optional[Sequence[Any]] = None) -> DataFrame:
         """
         Returns a new DataFrame representing the results of a SQL query.
-        You can use this method to execute a SQL statement. Note that you still
-        need to call :func:`DataFrame.collect` to execute this query in Snowflake.
+
+        **Note:** You can use this method to execute a SQL query, however the method is of **lazy evaluation**
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/session.py:1917]:
```suggestion
        **Note:** You can use this method to execute a SQL query lazily,
```

File: src/snowflake/snowpark/session.py [hunk 2]
```diff
@@ -1913,8 +1913,10 @@ def generator(
...
+         which means the sql is not executed immediately untill :func:`DataFrame.collect` is called.
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/session.py:1918]:
```suggestion
         which means the query is not executed until methods like :func:`DataFrame.collect` or :func:`DataFrame.to_pandas`  evaluates the DataFrame.
```

**Review by sfc-gh-stan — CHANGES_REQUESTED** [review]
Made two minor suggestions

File: src/snowflake/snowpark/session.py (hunk 2 above)
Comment (sfc-gh-mabrennan) [inline src/snowflake/snowpark/session.py:1918]: sql -> SQL

Comment (sfc-gh-mabrennan) [reply src/snowflake/snowpark/session.py:1918]: evaluates -> evaluate

**Review by sfc-gh-mabrennan — APPROVED** [review]
lgtm

File: src/snowflake/snowpark/session.py [hunk 3]
```diff
@@ -1913,8 +1913,12 @@ def generator(
     def sql(self, query: str, params: Optional[Sequence[Any]] = None) -> DataFrame:
         """
         Returns a new DataFrame representing the results of a SQL query.
-        You can use this method to execute a SQL statement. Note that you still
-        need to call :func:`DataFrame.collect` to execute this query in Snowflake.
+
+        Note:
+            You can use this method to execute a SQL query lazily,
+            which means the SQL is not executed untill methods like :func:`DataFrame.collect`
```
Comment (sfc-gh-mabrennan) [inline src/snowflake/snowpark/session.py:1919]: untill -> until

File: src/snowflake/snowpark/session.py [hunk 4]
```diff
@@ -1913,8 +1913,12 @@ def generator(
...
+            or :func:`DataFrame.to_pandas` evaluate the DataFrame.
+            For **immediate execution**, please chain the call with the collect method: `session.sql(query).collect()`.
```
Comment (sfc-gh-mabrennan) [inline src/snowflake/snowpark/session.py:1921]: remove "please".

**Review by sfc-gh-stan — APPROVED** [review]

**Pull request merged**